### PR TITLE
signalfx: remove unnecessary allocation if the distribution summary does not have histogram

### PR DESCRIPTION
**Pull request by bogdandrutu**
This optimization already in place for Timer.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java [hunk 1]
```diff
@@ -54,7 +54,8 @@ final class SignalfxDistributionSummary extends AbstractDistributionSummary {
         super(id, clock, CumulativeHistogramConfigUtil.updateConfig(distributionStatisticConfig), scale, false);
         this.countTotal = new StepTuple2<>(clock, stepMillis, 0L, 0.0, count::sumThenReset, total::sumThenReset);
         max = new TimeWindowMax(clock, distributionStatisticConfig);
-        if (isDelta) {
+        if (!distributionStatisticConfig.isPublishingPercentiles()
+                && distributionStatisticConfig.isPublishingHistogram() && isDelta) {
```
Comment (jonatan-ivanov) [inline implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java:58]: Is it possible that someone wants to publish both percentiles and histogram (for any reason)?
In that case, I would do 
```suggestion
        if (distributionStatisticConfig.isPublishingHistogram() && isDelta) {
```

Comment (bogdandrutu) [reply implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java:58]: If that is a possibility, though the current AbstractTimer does not do that, I think your suggestion is good. will change the timer as well.

Comment (bogdandrutu) [reply implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java]: Done.

**Review by jonatan-ivanov — APPROVED** [review]

**Comment by jonatan-ivanov** [comment]
Thank you! :)

**Pull request merged**